src/DynamoCore/Graph/Workspaces/HomeWorkspaceModel.cs

```diff
@@ -186,9 +205,19 @@ internal bool IsEvaluationPending
         public virtual void OnEvaluationStarted(EventArgs e)
         {
             this.HasRunWithoutCrash = false;
+            GraphRunInProgress = true;
 
-            var handler = EvaluationStarted;
-            if (handler != null) handler(this, e);
+            try
+            {
+                // Do not allow graph runs to be trigered from the EvaluationStarted event.
+                // Use a simple internal flag like ForceBlockRun (RunEnabled would notify WPF of changes and that would incur peformance penalties without any benefit)
+                RunSettings.ForceBlockRun = true;
```

Comment:
couldn't you use `GraphRunInProgress` that you just added?

Reply:
`GraphRunInProgress` does not block execution.
It is just a flag that indicates whether Dynamo is currently running a graph.

Reply:
We are now consistently blocking calls to `Run()` (graph run) from the `EvaluationStarted` event for the following reasons:
1. We did this before my PR, only it was not consistent
2. Allowing subsequent calls to Run() from `EvaluationStarted` can lead to infinite loops
3. When the `EvaluationStarted` event is fired, we are already inside the `Run` function. Node modifications are allowed (ex one can add more nodes to be computed in the `EvaluationStarted` event)

Clarifications on 1 (We did this before my PR):
Only calls to RequestRun were blocked by 
```
if ((RunSettings.RunEnabled || executingTask) && !DelayGraphExecution)
                {
                    Run();
                }
```
Depending on the timing of things, we could have had `executingTask` be true or false during the `EvaluationStarted` event, because `executingTask` is set to true before the tasks are scheduled and set back to false after a task is completed (asynchronously)